packages/peregrine/lib/talons/PageLoadingIndicator/usePageLoadingIndicator.js

```diff
@@ -0,0 +1,40 @@
+import { useEffect, useRef, useState } from 'react';
+import { useAppContext } from '../../context/app';
+
+export default () => {
+    const [{ isPageLoading }] = useAppContext();
+    const doneTimeoutRef = useRef();
+    const [loadingState, setLoadingState] = useState('off');
+
+    useEffect(() => {
+        if (isPageLoading) {
+            setLoadingState('loading');
+        } else if (
+            typeof clearTimeout !== 'undefined' &&
+            typeof setTimeout !== 'undefined'
+        ) {
+            setLoadingState('done');
+
+            if (doneTimeoutRef && doneTimeoutRef.current !== null) {
+                clearTimeout(doneTimeoutRef.current);
+            }
+
+            doneTimeoutRef.current = setTimeout(() => {
+                setLoadingState(currentLoadingState => {
+                    if (currentLoadingState === 'loading') {
+                        return 'loading';
+                    }
+
+                    return 'off';
+                });
+            }, 750);
```

Comment:
If this is a magic number, we should at least extract it to a constant within this file, if not an environment variable or something.

Reply:
Good call! I changed it to a constant higher up in the file. Basically it serves to group together loadings that may occur right after eachother so we don't see a loading bar that bounces back and forth